pkg/operator/cluster/ceph/mon/mon.go

```diff
@@ -134,61 +107,61 @@ func (c *Cluster) Start() error {
 		return fmt.Errorf("failed to initialize ceph cluster info. %+v", err)
 	}
 
-	if len(c.clusterInfo.Monitors) < c.Size {
-		c.startMons()
-	} else {
-		// Check the health of a previously started cluster
-		if err := c.checkHealth(); err != nil {
-			logger.Warningf("failed to check mon health %+v. %+v", c.clusterInfo.Monitors, err)
-		}
-	}
-
-	return nil
+	return c.startMons()
 }
 
 func (c *Cluster) startMons() error {
-	// init the mons config
-	mons := c.initMonConfig(c.Size)
+	// create statefulset with replicas 0
+	c.startStatefulSet(0)
 
-	// Assign the pods to nodes
-	if err := c.assignMons(mons); err != nil {
-		return fmt.Errorf("failed to assign pods to mons. %+v", err)
-	}
+	mons := []*monConfig{}
 
-	// Start one monitor at a time
+	// Init the mons config
 	for i := len(c.clusterInfo.Monitors); i < c.Size; i++ {
-		// Init the mon IPs
-		if err := c.initMonIPs(mons[0 : i+1]); err != nil {
-			return fmt.Errorf("failed to init mon services. %+v", err)
-		}
+		monName := getMonNameForID(i)
 
-		// save the mon config after we have "initiated the IPs"
-		if err := c.saveMonConfig(); err != nil {
+		c.clusterInfo.MonMutex.Lock()
```

Comment:
i would add a `defer` here instead of adding it after. Looks clear and ensures this is called if the logic path doesnt have an `unlock`